cloudinit/config/cc_growpart.py

```diff
@@ -496,7 +496,12 @@ def resize_devices(resizer, devices, distro: Distro):
             continue
 
         LOG.debug("growpart found fs=%s", fs)
-        if fs == "zfs":
+        # TODO: This seems to be the wrong place for this. On Linux, we the
+        # `os.stat(blockdev)` call below will fail on a ZFS filesystem.
+        # We then delay resizing the FS until calling cc_resizefs. Yet
+        # the code here is to accommodate the FreeBSD `growfs` service.
+        # Ideally we would grow the FS for both OSes in the same module.
+        if fs == "zfs" and isinstance(resizer, ResizeGrowFS):
             info += _call_resizer(resizer, devent, disk, ptnum, blockdev, fs)
             return info
```

Comment:
This should be a continue, not a return in case there are other devices listed in `growpart: devices: [..]`
```suggestion
            continue
```